vanilla-tests/src/main/java/fixtures/discriminatorenum/models/Dog.java

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package fixtures.discriminatorenum.models;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.json.JsonReader;
+import com.azure.json.JsonSerializable;
+import com.azure.json.JsonToken;
+import com.azure.json.JsonWriter;
+import java.io.IOException;
+
+/**
+ * Test extensible enum type for discriminator.
+ */
+@Fluent
+public class Dog implements JsonSerializable<Dog> {
+    /*
+     * discriminator property
+     */
+    private DogKind kind;
+
+    /*
+     * Weight of the dog
+     */
+    private int weight;
+
+    /**
+     * Creates an instance of Dog class.
+     */
+    public Dog() {
+        this.kind = DogKind.fromString("Dog");
+    }
+
+    /**
+     * Get the kind property: discriminator property.
+     * 
+     * @return the kind value.
+     */
+    public DogKind getKind() {
+        return this.kind;
+    }
```

Comment:
We can see, even with Swagger, the return value is still an enum (when Swagger defines it as enum).